### ISSUE TITLE
pip install Fails In Python 3.10

### ISSUE THREAD

**Issue by jmcclare** [ccsb-scripps/AutoDock-Vina#176]
Sometime after Jan. 27, 2023 using pip to install vina 1.2.3 in a Python 3.10 environment stopped working.

Up until then PyPI must have had a pre‐built wheel file for Python 3.10. Now there is no Python 3.10 wheel file, so pip and setuptools attempt to build the vina Python package and they fail with the following error:

```
error in vina setup command: 'python_requires' must be a string containing valid version specifiers; Invalid specifier: '>=3.5.*'
```

In [`build/python/setup.py`, line 361](https://github.com/ccsb-scripps/AutoDock-Vina/blob/develop/build/python/setup.py#L361), we need to change:

```python
    python_requires='>=3.5.*',
```

To:

```python
    python_requires='>=3.5',
```

In the [setuptools code](https://github.com/pypa/setuptools/blob/main/setuptools/_vendor/packaging/specifiers.py#L187) it states in the comments that prefix matching wildcards are not allowed in version specifiers with operators like `>=`.

A short term fix would be to upload a Python 3.10 wheel file for vina 1.2.3 to PyPI so that pip would not have to build the package.

## Steps to reproduce ##

In Ubuntu 22.04

```bash
sudo apt-get install -y python3.10 python3.10-venv python3.10-dev build-essential cmake swig

python3.10 -m venv vina-venv
source vina-venv/bin/activate

pip install --no-cache-dir 'vina==1.2.3'
```

Output:

```
Collecting vina==1.2.3
  Downloading vina-1.2.3.tar.gz (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 kB 2.8 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [3 lines of output]
      fatal: not a git repository (or any of the parent directories): .git
      Version found 1.2.3 (from __init__.py)
      error in vina setup command: 'python_requires' must be a string containing valid version specifiers; Invalid specifier: '>=3.5.*'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

No combination of upgrading, or downgrading pip, setuptools, or wheel will avoid this issue.

**Comment by Evert-Homan** [comment]
I have a similar problem, also Ubuntu 22.04, with Python 3.10.6:

```
evehom`  (e) vina  /  opt  pip install vina                                 
Collecting vina                                                                 
  Downloading vina-1.2.3.tar.gz (95 kB)                                         
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 kB 1.9 MB/s eta 0:00:00
  Installing build dependencies ... done                                        
  Getting requirements to build wheel ... error                                 
  error: subprocess-exited-with-error                                           
                                                                                
  × Getting requirements to build wheel did not run successfully.               
  │ exit code: 1                                                                
  ╰─> [3 lines of output]                                                       
      fatal: not a git repository (or any of the parent directories): .git      
      Version found 1.2.3 (from __init__.py)                                    
      error in vina setup command: 'python_requires' must be a string containing valid version specifiers; Invalid specifier: '>=3.5.*'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Please advise on how to resolve this. Thanks/Evert

**Comment by JackBettam** [comment]
Is there any update on this - currently running into the exact same issue.

**Comment by 2024sd552** [comment]
Same error, but one line different.


Requirement already satisfied: numpy in c:\users\sdaha\miniconda3\envs\vina\lib\site-packages (1.22.3)
Collecting numpy
  Using cached numpy-1.24.3-cp310-cp310-win_amd64.whl (14.8 MB)
Collecting vina
  Using cached vina-1.2.3.tar.gz (95 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [2 lines of output]
      Version found 1.2.3 (from __init__.py)
      error in vina setup command: 'python_requires' must be a string containing valid version specifiers; Invalid specifier: '>=3.5.*'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by JackBettam** [comment]
> Same error, but one line different.

If you haven't already, a workaround is to change your Python version in the environment. I used 3.9.7 and it seemed to work.

**Comment by JackBettam** [comment]
@2024sd552 when you changed the Python version was this a new environment when you first install Python? Or do you have to uninstall Python and then reinstall Python? 

My advice would be to follow the tutorial to install in AutoDock Vina but replace the line of code below in the installation guide:

```
Replace this:
$ conda create -n vina python=3
With this:
$ conda create -n vina python=3.9
```

As far as I know the issue is that the version check >=3.5 fails as 3.10 is a number that technically is below 3.5. By using the latest 3.9 Python it should still build. 

Hopefully the next release will fix this version check issue. If you're still having issues, make a clean environment, and try again. If it doesn't work I've got a tutorial I did for myself in case I needed to reinstall any of the docking tools, so let me know if it still doesn't work and I'll send that to you.

**Comment by 2024sd552** [comment]
> @2024sd552 when you changed the Python version was this a new environment when you first install Python? Or do you have to uninstall Python and then reinstall Python?
> 
> My advice would be to follow the tutorial to install in AutoDock Vina but replace the line of code below in the installation guide:
> 
> ```
> Replace this:
> $ conda create -n vina python=3
> With this:
> $ conda create -n vina python=3.9
> ```
> 
> As far as I know the issue is that the version check >=3.5 fails as 3.10 is a number that technically is below 3.5. By using the latest 3.9 Python it should still build.
> 
> Hopefully the next release will fix this version check issue. If you're still having issues, make a clean environment, and try again. If it doesn't work I've got a tutorial I did for myself in case I needed to reinstall any of the docking tools, so let me know if it still doesn't work and I'll send that to you.

Hey I tried what you said and still am running into the same issue. I even tried downgrading my miniconda's system python version to 3.9 and it still wasn't working. I was wondering if you could share what you did as I am so lost on what to do at this point.

**Comment by JackBettam** [comment]
Hey - I have made the repo I was uploading my instructions to public: [https://github.com/JackBettam/LinuxTest](https://github.com/JackBettam/LinuxTest)

I was using WSL2 running Ubuntu. I hope it helps - it seems to work fine when I'm running it now, so hope it will work okay for you.

**Comment by SAhn97** [comment]
having the same issue here

**Comment by diogomart** [comment]
Fixed in #167
